code/scripting/global_hooks.cpp

```diff
@@ -1,23 +1,108 @@
+	{ {"Action", "string", "The name of the action that was executed."} });
+
+const std::shared_ptr<Hook<ControlActionConditions>> OnActionStopped = Hook<ControlActionConditions>::Factory("On Action Stopped",
+	"Invoked whenever a user action is no longer invoked through control input.",
+	{ {"Action", "string", "The name of the action that was stopped."} });
+
+const std::shared_ptr<Hook<>> OnKeyPressed = Hook<>::Factory("On Key Pressed",
+	"Invoked whenever a key is pressed.",
+	{ {"Key", "string", "The scancode of the key that has been pressed."} });
+
+const std::shared_ptr<Hook<>> OnKeyReleased = Hook<>::Factory("On Key Released",
+	"Invoked whenever a key is released.",
+	{ {"Key", "string", "The scancode of the key that has been released."} });
+
+const std::shared_ptr<Hook<>> OnMouseMoved = Hook<>::Factory("On Mouse Moved",
+	"Invoked whenever the mouse is moved.",
+	{});
+
+const std::shared_ptr<Hook<>> OnMousePressed = Hook<>::Factory("On Mouse Pressed",
+	"Invoked whenever a mouse button is pressed.",
+	{});
+
+const std::shared_ptr<Hook<>> OnMouseReleased = Hook<>::Factory("On Mouse Released",
+	"Invoked whenever a mouse button is released.",
+	{});
+
+const std::shared_ptr<Hook<ShipSourceConditions>> OnAfterburnerStart = Hook<ShipSourceConditions>::Factory("On Afterburner Engage",
+	"Invoked whenever a ship engages its afterburners",
+	{ { "Ship", "ship", "The ship engaging its afterburners" } });
+
+const std::shared_ptr<Hook<ShipSourceConditions>> OnAfterburnerEnd = Hook<ShipSourceConditions>::Factory("On Afterburner Stop",
+	"Invoked whenever a ship stops using its afterburners",
+	{ { "Ship", "ship", "The ship which had been using its afterburners" } });
+
+const std::shared_ptr<Hook<ShipSourceConditions>> OnWaypointsDone = Hook<ShipSourceConditions>::Factory("On Waypoints Done",
+	"Invoked whenever a ship stops using its afterburners",
+	{
+		{ "Ship", "ship", "The ship which has completed the waypoints." },
+		{ "Wing", "wing", "The wing which the ship belongs to. Can be invalid." },
+		{ "Waypointlist", "waypointlist", "The set of waypoints which was completed." }
+	});
+
+const std::shared_ptr<Hook<ShipSourceConditions>> OnGoalsCleared = Hook<ShipSourceConditions>::Factory("On Goals Cleared",
+	"Invoked whenever a ship has its goals cleared.",
+	{ { "Ship", "ship", "The ship whose goals are cleared." } });
+
+const std::shared_ptr<Hook<ShipSourceConditions>> OnDebrisCreated = Hook<ShipSourceConditions>::Factory(
 	"On Debris Created",
 	"Invoked when a piece of debris is created.",
 	{
 		{"Debris", "debris", "The newly created debris object"},
 		{"Source", "object", "The object (probably a ship) from which this debris piece was spawned."},
 	});
 
-const std::shared_ptr<OverridableHook> OnShipCollision = OverridableHook::Factory("On Ship Collision",
+const std::shared_ptr<OverridableHook<CollisionConditions>> OnShipCollision = OverridableHook<CollisionConditions>::Factory("On Ship Collision",
```

Comment:
on the "Hitpos" line (for some reason GitHub isn't letting me select that specific line), "collision" is misspelled